lib/Serialization/Deserialization.cpp

```diff
@@ -166,33 +162,15 @@ static void skipRecord(llvm::BitstreamCursor &cursor, unsigned recordKind) {
   (void)kind;
 }
 
-void ModuleFile::fatal(llvm::Error error) {
-  if (FileContext) {
-    getContext().Diags.diagnose(SourceLoc(), diag::serialization_fatal, Core->Name);
-    getContext().Diags.diagnose(
-        SourceLoc(), diag::serialization_misc_version, Core->Name,
-        Core->MiscVersion, allowCompilerErrors());
-
-    if (!Core->CompatibilityVersion.empty()) {
-      if (getContext().LangOpts.EffectiveLanguageVersion
-            != Core->CompatibilityVersion) {
-        SmallString<16> effectiveVersionBuffer, compatVersionBuffer;
-        {
-          llvm::raw_svector_ostream out(effectiveVersionBuffer);
-          out << getContext().LangOpts.EffectiveLanguageVersion;
-        }
-        {
-          llvm::raw_svector_ostream out(compatVersionBuffer);
-          out << Core->CompatibilityVersion;
-        }
-        getContext().Diags.diagnose(
-            SourceLoc(), diag::serialization_compatibility_version_mismatch,
-            effectiveVersionBuffer, Core->Name, compatVersionBuffer);
```

Comment:
Do we still have the information about the Swift language compatibility version of the module somewhere in the output? This note has been useful to me at least once to quickly explain a deserialization failure due to changes in ClangImporter behaviors between two Swift versions.

Reply:
See above re. the effective version, as an example output:
```
Stack dump:
0.	Program arguments: ......
1.	Swift version 5.5-dev (LLVM <hash>, Swift <hash>)
2.	Compiling with Swift version 5.5-dev effective-4.2 (LLVM <hash>, Swift <hash>)
3.	While evaluating request TypeCheckSourceFileRequest(source_file "<stdin>")
4.	While type-checking statement at [<stdin>:1:13 - line:1:21] RangeText="_ = Sub."
5.	While type-checking expression at [<stdin>:1:13 - line:1:21] RangeText="_ = Sub."
6.	While evaluating request QualifiedLookupRequest(0x7fa27c81ce00 TopLevelCodeDecl line=1, {Lib.(file).Sub}, 'disappearingMethod', { NL_ProtocolMembers, NL_RemoveNonVisible, NL_RemoveOverridden, NL_OnlyTypes })
7.	While evaluating request DirectLookupRequest(directly looking up 'disappearingMethod' on Lib.(file).Sub with options {  })
8.	While loading members for 'Sub' (in module 'Lib')
9.	*** DESERIALIZATION FAILURE ***
module 'Lib' with full misc version '5.5(4.1.50)/Swift version 5.5-dev (LLVM <hash>, Swift <hash>)'
could not find 'disappearingMethod()' in parent class
```

So here we can see the version being compiled is `effective-4.2` and the module was compiled with `(4.1.50)`